NDTensors/src/tensorstorage/tensorstorage.jl

```diff
@@ -28,6 +28,7 @@ Base.@propagate_inbounds function Base.setindex!(S::TensorStorage, v, i::Integer
   return (setindex!(data(S), v, i); S)
 end
 
+## Missing a check or conversion when calling number * Tensor. This causes Metal to fail numerically because it tries to convert it to Float64. Preserve S eltype. ## TODO this could probably be handled differently/better?
```

Comment:
This line can fail on Metal GPU when x is a Float64 because I believe there is an internal type promotion. I am not currently on my mac and will send code when I am. Could this type of operation cause the code to fail when, for example, we are using multi-float?

Reply:
That's just how Julia's type promotion works:
```julia
julia> one(Float64) * rand(Float32, 2)
2-element Vector{Float64}:
 0.9090054631233215
 0.7104539275169373
```
So whatever issue you are seeing, the primary problem is that a `Float64` is being sent down to this code, this code looks ok to me.

Reply:
@mtfishman a quick example on CUDA shows this 
```
julia> v = CuVector{Float32}(undef, 10)
10-element CuArray{Float32, 1, CUDA.Mem.DeviceBuffer}:

julia> v * 2.0
10-element CuArray{Float64, 1, CUDA.Mem.DeviceBuffer}:
```

Reply:
Right that all makes sense, thanks!
I guess the issue I am finding is on line 77 in `NDTensors/test/dense.jl` 
```
data(A * 2.0) == data(2.0 * A)
```
since metal uses a `Float32`, this line tries to promote the type up to  `Float64` but for the most part in ITensors/NDTensors we aren't multiplying a our tensors by random number without considering the eltype of `A` and the scalar.  So in this test I will wrap the number in `elt(2.0)` to ensure that it preserves eltype.